### PR TITLE
feat: Adding calls to custom style hooks derived from context in react-checkbox components

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Changes
- Updated components to call custom style hook

## Issues
- Updates #27058

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                     |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                      Change |
| :-------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr>                      |  `33.091 kB`<br />`10.319 kB` |  `35.856 kB`<br />`10.821 kB` | `2.765 kB` <br />`502 B`  |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/CheckboxField.fixture.js'>CheckboxField</abbr>            |  `39.996 kB`<br />`12.206 kB` |  `42.761 kB`<br />`12.701 kB` | `2.765 kB` <br />`495 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                         | `144.785 kB`<br />`39.447 kB` |  `147.55 kB`<br />`39.934 kB` | `2.765 kB` <br />`487 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>         |   `134.4 kB`<br />`34.514 kB` | `137.167 kB`<br />`35.234 kB` | `2.767 kB` <br />`720 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr> |  `82.316 kB`<br />`20.576 kB` |  `85.081 kB`<br />`21.315 kB` | `2.765 kB` <br />`739 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>           |  `81.637 kB`<br />`20.386 kB` |  `84.402 kB`<br />`21.128 kB` | `2.765 kB` <br />`742 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `67.061 kB`<br />`18.105 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `205.644 kB`<br />`57.056 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `37.891 kB`<br />`12.039 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.324 kB`<br />`2.129 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `43.776 kB`<br />`12.063 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/efb2274e37c0237f3a10a4630b2696a9880f6398'>efb2274e37c0237f3a10a4630b2696a9880f6398</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/Avatar-mount_base.html">1292</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/Avatar-mount.html">1290</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/Button-mount_base.html">932</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/Button-mount.html">944</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/Field-mount_base.html">1947</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/Field-mount.html">1924</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/FluentProvider-mount_base.html">1550</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/FluentProvider-mount.html">1535</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/FluentProviderWithTheme-mount_base.html">587</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/FluentProviderWithTheme-mount.html">600</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">543</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">546</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">581</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">586</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/InfoButton-mount_base.html">520</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/InfoButton-mount.html">515</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/MakeStyles-mount_base.html">1975</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/MakeStyles-mount.html">1941</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/Persona-mount_base.html">2862</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/Persona-mount.html">2820</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/SpinButton-mount_base.html">2329</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27062/perf-test-react-components/SpinButton-mount.html">2253</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: change/@fluentui-react-checkbox-dae70264-8248-4616-bd21-87174e312d3d.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "feat: custom styles",
```
Comment (khmakoto) [inline change/@fluentui-react-checkbox-dae70264-8248-4616-bd21-87174e312d3d.json:3]:
```suggestion
  "comment": "feat: Adding calls to custom style hooks derived from context.",
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**